pkg/cluster/resources.go

```diff
@@ -128,6 +132,95 @@ func (c *Cluster) preScaleDown(newStatefulSet *v1beta1.StatefulSet) error {
 	return nil
 }
 
+// setRollingUpdateFlagForStatefulSet sets the indicator or the rolling upgrade requirement
```

Comment:
I don't think "reuse" is the right word here; essentially, we set the annotation upon determining that the rolling update is necessary and don't clear it until the rolling update finishes successfully.  Normally, the rolling update should happen immediately after the label is set, as both should occur inside the `syncStatefulSet` method. However, in case the update goes sideways or the operator dies in progress, the label serves as an indicator that the new instance of the operator should retry the update; therefore, we can clear the label only after the update succeeds.